versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapter.java

```diff
@@ -229,26 +249,32 @@ private void delete(MongoCollection<Document> collection, Set<Hash> ids) {
     }
   }
 
+  private <T> List<T> fetchMappedPage(
+      MongoCollection<Document> collection, List<Hash> hashes, Function<Document, T> mapper) {
+    return fetchMappedPageGeneric(
+        collection, hashes.stream().map(Hash::asString).collect(Collectors.toList()), mapper);
+  }
+
   /**
    * Fetches a collection of documents by hash and returns them in the order of hashes requested.
    *
    * <p>Note: unknown hashes will correspond to {@code null} elements in the result list.
    */
-  private <T> List<T> fetchMappedPage(
-      MongoCollection<Document> collection, List<Hash> hashes, Function<Document, T> mapper) {
-    List<Document> ids = hashes.stream().map(this::toId).collect(Collectors.toList());
+  private <T> List<T> fetchMappedPageGeneric(
+      MongoCollection<Document> collection, List<String> idList, Function<Document, T> mapper) {
```

Comment:
why do we get rid of type safety here (using `Hash`) and starting using `String` instead?
I see that the caller has a hash and transforms it to `String` (`Hash::asString`). 
Shouldn't we favor strong type instead of just passing `String`?